hack/test/e2e-iso.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+set -eou pipefail
+
+source ./hack/test/e2e.sh
+
+PROVISIONER=qemu
+CLUSTER_NAME=e2e-iso
+
+function create_cluster {
+  build_registry_mirrors
+
+  "${TALOSCTL}" cluster create \
+    --iso-path=${ARTIFACTS}/talos-amd64.iso \
+    --skip-injecting-config \
+    --wait=false \
+    --provisioner "${PROVISIONER}" \
+    --name "${CLUSTER_NAME}" \
+    --masters=1 \
+    --workers=0 \
+    --mtu 1500 \
+    --memory 2048 \
+    --cpus 2.0 \
+    --cidr 172.20.2.0/24 \
+    --install-image ${REGISTRY:-ghcr.io}/talos-systems/installer:${INSTALLER_TAG} \
+    --cni-bundle-url ${ARTIFACTS}/talosctl-cni-bundle-'${ARCH}'.tar.gz \
+    --crashdump \
+    ${REGISTRY_MIRROR_FLAGS}
+
+  "${TALOSCTL}" config node 172.20.2.2
+}
+
+function destroy_cluster() {
+  "${TALOSCTL}" cluster destroy --name "${CLUSTER_NAME}" --provisioner "${PROVISIONER}"
+}
+
+create_cluster
+timeout -v --preserve-status 1m  bash -c "until ${TALOSCTL} apply-config -n 127.0.0.1 --insecure -f controlplane.yaml; do sleep 5; done"
+get_kubeconfig
+run_talos_integration_test
```

Comment:
instead of running full test, we could do `talosctl -n 172.20.0.2 cluster health` to save some time in CI, this would only verify bootstrap, but that should be enough?